core/types/src/macros.rs

```diff
@@ -42,6 +42,27 @@ macro_rules! new_type_accessors_impls {
     )*}
 }
 
+/// Newtype wrapper for a primitive or struct type
+#[macro_export]
+macro_rules! impl_newtype {
+    ($($wrapper:ident, $inner:ty;)*) => {$(
+        $crate::newtype_accessors_impls! {
+            $wrapper, $inner;
+        }
+
+        impl core::fmt::Display for$wrapper {
+            fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+                if f.alternate() {
+                    write!(f, "{:#?}", self.0)
+                } else {
+                    write!(f, "{:?}", self.0)
```

Comment:
Not fond of re-directing to debug here, but it seemed like a nicer display than the struct display would be a case by case basis, and should be handled as they arise.